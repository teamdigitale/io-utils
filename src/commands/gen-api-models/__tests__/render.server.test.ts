import { OpenAPIV2 } from "openapi-types";
import * as SwaggerParser from "swagger-parser";

import { parseAllOperations } from "../parse";
import { renderServerCode } from "../render";
describe("gen-api-models - server", () => {
  let spec: OpenAPIV2.Document;
  beforeAll(
    async () =>
      (spec = (await SwaggerParser.bundle(
        `${process.cwd()}/__mocks__/api.yaml`
      )) as OpenAPIV2.Document)
  );

  it("should render a client", async () => {
    const allOperations = parseAllOperations(spec, "undefined", "undefined");
    const code = await renderServerCode(spec, allOperations);

    expect(code).toMatchSnapshot();
  });
});
