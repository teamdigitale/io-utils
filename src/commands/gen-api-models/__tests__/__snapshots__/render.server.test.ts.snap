// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gen-api-models - server should render a client 1`] = `
"/**
 * Do not edit this file it is auto-generated by io-utils / gen-api-models.
 * See https://github.com/pagopa/io-utils
 */
/* eslint-disable  */

export interface ITestAuthBearerRequestParams {
  qo?: string;

  qr: string;

  cursor?: string;
}

export type ITestAuthBearerRequestHandlerParams<
  OtherParams extends Record<string, K>,
  K
> = ITestAuthBearerRequestParams & OtherParams;

type Success =
  /* 200 0 */
  IResponseSuccessJson<undefined>;

{
}

export type ITestAuthBearerRequestHandler<
  OtherParams extends Record<string, K>,
  K
> = (
  args: ITestAuthBearerRequestHandlerParams<OtherParams, K>
) => Promise<Success | Failures>;

export interface ITestSimpleTokenRequestParams {
  qo?: string;

  qr: string;

  cursor?: string;
}

export type ITestSimpleTokenRequestHandlerParams<
  OtherParams extends Record<string, K>,
  K
> = ITestSimpleTokenRequestParams & OtherParams;

type Success =
  /* 200 0 */
  IResponseSuccessJson<undefined>;

{
}

export type ITestSimpleTokenRequestHandler<
  OtherParams extends Record<string, K>,
  K
> = (
  args: ITestSimpleTokenRequestHandlerParams<OtherParams, K>
) => Promise<Success | Failures>;

export interface ITestMultipleSuccessRequestParams {}

export type ITestMultipleSuccessRequestHandlerParams<
  OtherParams extends Record<string, K>,
  K
> = ITestMultipleSuccessRequestParams & OtherParams;

type Success =
  /* 200 0 */
  | IResponseSuccessJson<Message>
  /* 202 1 */
  | ResponseSuccessAccepted<undefined>;

{
}

export type ITestMultipleSuccessRequestHandler<
  OtherParams extends Record<string, K>,
  K
> = (
  args: ITestMultipleSuccessRequestHandlerParams<OtherParams, K>
) => Promise<Success | Failures>;

export interface ITestFileUploadRequestParams {
  file: { uri: string; name: string; type: string };
}

export type ITestFileUploadRequestHandlerParams<
  OtherParams extends Record<string, K>,
  K
> = ITestFileUploadRequestParams & OtherParams;

type Success =
  /* 200 0 */
  IResponseSuccessJson<undefined>;

{
}

export type ITestFileUploadRequestHandler<
  OtherParams extends Record<string, K>,
  K
> = (
  args: ITestFileUploadRequestHandlerParams<OtherParams, K>
) => Promise<Success | Failures>;

export interface ITestResponseHeaderRequestParams {}

export type ITestResponseHeaderRequestHandlerParams<
  OtherParams extends Record<string, K>,
  K
> = ITestResponseHeaderRequestParams & OtherParams;

type Success =
  /* 201 0 */
  IResponseSuccessRedirectToResource<Message, Message>;

{
}

export type ITestResponseHeaderRequestHandler<
  OtherParams extends Record<string, K>,
  K
> = (
  args: ITestResponseHeaderRequestHandlerParams<OtherParams, K>
) => Promise<Success | Failures>;

export interface ITestParameterWithReferenceRequestParams {
  createdMessage?: Message;
}

export type ITestParameterWithReferenceRequestHandlerParams<
  OtherParams extends Record<string, K>,
  K
> = ITestParameterWithReferenceRequestParams & OtherParams;

type Success =
  /* 201 0 */
  IResponseSuccessRedirectToResource<undefined, undefined>;

{
}

export type ITestParameterWithReferenceRequestHandler<
  OtherParams extends Record<string, K>,
  K
> = (
  args: ITestParameterWithReferenceRequestHandlerParams<OtherParams, K>
) => Promise<Success | Failures>;

export interface ITestParameterWithDashRequestParams {
  pathParam?: string;

  fooBar?: string;

  requestId?: string;
}

export type ITestParameterWithDashRequestHandlerParams<
  OtherParams extends Record<string, K>,
  K
> = ITestParameterWithDashRequestParams & OtherParams;

type Success =
  /* 200 0 */
  IResponseSuccessJson<undefined>;

{
}

export type ITestParameterWithDashRequestHandler<
  OtherParams extends Record<string, K>,
  K
> = (
  args: ITestParameterWithDashRequestHandlerParams<OtherParams, K>
) => Promise<Success | Failures>;

export interface ITestParameterWithDashAnUnderscoreRequestParams {
  pathParam?: string;

  fooBar?: string;

  requestId?: string;
}

export type ITestParameterWithDashAnUnderscoreRequestHandlerParams<
  OtherParams extends Record<string, K>,
  K
> = ITestParameterWithDashAnUnderscoreRequestParams & OtherParams;

type Success =
  /* 200 0 */
  IResponseSuccessJson<undefined>;

{
}

export type ITestParameterWithDashAnUnderscoreRequestHandler<
  OtherParams extends Record<string, K>,
  K
> = (
  args: ITestParameterWithDashAnUnderscoreRequestHandlerParams<OtherParams, K>
) => Promise<Success | Failures>;

export interface ITestWithTwoParamsRequestParams {
  firstParam?: string;

  secondParam?: string;
}

export type ITestWithTwoParamsRequestHandlerParams<
  OtherParams extends Record<string, K>,
  K
> = ITestWithTwoParamsRequestParams & OtherParams;

type Success =
  /* 200 0 */
  IResponseSuccessJson<undefined>;

{
}

export type ITestWithTwoParamsRequestHandler<
  OtherParams extends Record<string, K>,
  K
> = (
  args: ITestWithTwoParamsRequestHandlerParams<OtherParams, K>
) => Promise<Success | Failures>;
"
`;
